internal/controller/humioaction_controller.go

```diff
@@ -319,198 +318,222 @@ func (r *HumioActionReconciler) logErrorAndReturn(err error, msg string) error {
 // actionAlreadyAsExpected compares fromKubernetesCustomResource and fromGraphQL. It returns a boolean indicating
 // if the details from GraphQL already matches what is in the desired state of the custom resource.
 // If they do not match, a map is returned with details on what the diff is.
-// nolint:gocyclo
 func actionAlreadyAsExpected(expectedAction humiographql.ActionDetails, currentAction humiographql.ActionDetails) (bool, map[string]string) {
-	diffMap := map[string]string{}
-	actionType := "unknown"
-	redactedValue := "<redacted>"
+	diffMap := compareActions(expectedAction, currentAction)
+	actionType := getActionType(expectedAction)
+
+	diffMapWithTypePrefix := addTypePrefix(diffMap, actionType)
+	return len(diffMapWithTypePrefix) == 0, diffMapWithTypePrefix
+}
 
-	switch e := (expectedAction).(type) {
+func getActionType(action humiographql.ActionDetails) string {
+	switch action.(type) {
 	case *humiographql.ActionDetailsEmailAction:
-		switch c := (currentAction).(type) {
-		case *humiographql.ActionDetailsEmailAction:
-			actionType = getTypeString(e)
-			if diff := cmp.Diff(c.GetName(), e.GetName()); diff != "" {
-				diffMap["name"] = diff
-			}
-			if diff := cmp.Diff(c.GetRecipients(), e.GetRecipients()); diff != "" {
-				diffMap["recipients"] = diff
-			}
-			if diff := cmp.Diff(c.GetSubjectTemplate(), e.GetSubjectTemplate()); diff != "" {
-				diffMap["subjectTemplate"] = diff
-			}
-			if diff := cmp.Diff(c.GetEmailBodyTemplate(), e.GetEmailBodyTemplate()); diff != "" {
-				diffMap["bodyTemplate"] = diff
-			}
-			if diff := cmp.Diff(c.GetUseProxy(), e.GetUseProxy()); diff != "" {
-				diffMap["useProxy"] = diff
-			}
-		default:
-			diffMap["wrongType"] = fmt.Sprintf("expected type %T but current is %T", e, c)
-		}
+		return "email"
```

Comment:
Should we replace this 'hardcoded' switch with just `return getTypeString(action.(type))` or do we want a shorter more concise type?